### ISSUE TITLE
syntax_map doesn't work for "Gulpfile JS"

### ISSUE THREAD

**Issue by Uxspy** [SublimeLinter/SublimeLinter#449]
Hi,

I just found that in ST3, Sublimelinter doesn't work if the file's `language` was detected as `Gulpfile JS`.

By default, the Lint Mode was set to `background` and I found the Linter was not triggered, then I try to trigger it manually by select `Lint this view`, it doesn't work neither.

![image](https://cloud.githubusercontent.com/assets/3072295/15090194/3c63c3a0-1450-11e6-8251-d34c7dc3fc45.png)

**Comment by groteworld** [comment]
Hi and welcome!  What linter in particular were you trying to run? Eslint or JSHint? If either you will need to add Gulpfile JS to the syntax_map setting as JavaScript. Check out the SublimeLinter documentation on the syntax_map configuration option for an example. Thanks!

**Comment by Uxspy** [comment]
Thanks @groteworld , I'm using JSHint

**Comment by Uxspy** [comment]
Thanks @groteworld for your reminder, I just added "Gulpfile JS" to the syntax_map, however it doesn't work either, I guess this is because I'm using a theme which defined some .tmLanguages(Gulpfile JS defined in this theme package), however, there're always 2 .tmLanguages with the same name exists in the same theme package, I'm not sure if this is the reason why Linter still not work.

**Comment by jawshooah** [comment]
Is the file called `Gulpfile JS.tmLanguage`?

From the [docs](http://www.sublimelinter.com/en/latest/global_settings.html?highlight=syntax_map#syntax-map):

> Syntax names are the name of the `.tmLanguage` file that defines the syntax. This is not the file extension of the files to be linted, and may not necessarily be what is in the `View > Syntax` menu and in the lower right of the status bar.

**Comment by silverwind** [comment]
I also observe that adding `"Gulpfile JS"` to the `syntax_map` does not work. The string is exactly as in the filename. My theme defines various such languages:

````
Packages/Theme - Coffee/icons/Langs/Gruntfile Coffee.tmLanguage
Packages/Theme - Coffee/icons/Langs/Gruntfile JS.tmLanguage
Packages/Theme - Coffee/icons/Langs/Gulpfile Coffee.tmLanguage
Packages/Theme - Coffee/icons/Langs/Gulpfile JS.tmLanguage
````

Could it be that the space in the name throws SublimeLinter off?
